### ISSUE TITLE
Full-day event not showing on actual day of event.

### ISSUE THREAD

**Issue by evroom** [MMM-CalendarExt2/MMM-CalendarExt2#82]
module: MMM-CalendarExt2
MagicMirror: v2.11.0

Problem:

Full-day event not showing on actual day of event.

Examples:

Today is 26.05.2020, full event is on 27.05.2020, on 26.05.2020 it is showing on MM.
Today is 27.05.2020, full event is on 27.05.2020, on 27.05.2020 it is NOT showing on MM.

All events:

27.05.2020 - 18:00 - 19:00 - First entry for TestCal
27.05.2020 - fullday - FULLDAY_MAY_27
28.05.2020 - 18:00 - 19:00 - Second entry for TestCal
28.05.2020 - fullday - FULLDAY_MAY_28

On MM:

27.05.2020 - 18:00 - 19:00 - First entry for TestCal
28.05.2020 - 18:00 - 19:00 - Second entry for TestCal
28.05.2020 - fullday - FULLDAY_MAY_28

This one is missing:

27.05.2020 - fullday - FULLDAY_MAY_27

At 28.05.2020 - 00:00hrs this entry will disappear from MM:

28.05.2020 - fullday - FULLDAY_MAY_28

pm2 log:

```
0|MagicMir | [2020-05-27 13:14:13.657] [LOG]    [CALEXT2] calendar:Familie Calendar >> Scanning start with interval:60000
0|MagicMir | [2020-05-27 13:14:13.988] [LOG]    [CALEXT2] calendar:Familie Calendar >> Scanned: 4, Selected: 4
0|MagicMir | [2020-05-27 13:15:14.005] [LOG]    [CALEXT2] calendar:Familie Calendar >> Scanning start with interval:60000
0|MagicMir | [2020-05-27 13:15:14.336] [LOG]    [CALEXT2] calendar:Familie Calendar >> Scanned: 4, Selected: 4
```

That means parsing is correct.

Debugging shows that
~/MagicMirror/modules/MMM-CalendarExt2/node_helper.js
is parsing correct.

~/MagicMirror/config/config.js

Note: url has been editted.
The real one does not show any private information, but I do not know if it can be hacked.

```
  {
    module: 'MMM-CalendarExt2',
    disabled: false,
    config: {
      //rotateInterval: 20*1000, // Changing scene per this interval.
      rotateInterval: 0, // Disable scene rotate
      updateInterval: 1*60*1000, // Refreshing views interval.
      calendars : [
        {
          name: "Familie Calendar",
          scanInterval: 1*60*1000, // Scanning calendar interval.
          className: "googlecalendar_familie",
          url: "https://calendar.google.com/calendar/ical/TESTCAL_group.calendar.google.com/private-TESTCAL/basic.ics", // TestCal
        },
      ],
      scenes: [
        {
          name: "DEFAULT",
          views: ["Familie Calendar"],
        },
      ],
      views: [
        {
          name: "Familie Calendar",
          mode: "upcoming",
          position: "top_left",
          calendars: ["Familie Calendar"],
          slotCount: 1,
          slotTitle: "Family Calendar",
          slotMaxHeight: "150px",
          //filterPassedEvent: true,
          filterPassedEvent: false,
          maxItems: 12,
          //maxDays: 60,
          maxDays: 8,
          locale: "de-DE",
          hideOverflow: false,
          dateFormat: "ddd Do MMM",
          dateTimeFormat: {
            sameDay: "[Heute] HH:mm",
            nextDay: "[Morgen] HH:mm",
            nextWeek: "dddd HH:mm",
            lastDay: "[Gestern] HH:mm",
            lastWeek: "[Letzte] ddd HH:mm",
            sameElse: "ddd Do Mo HH:mm",
          },
          timeFormat: "HH:mm",
          relativeFormat: {
            passed: "ended %ENDFROMNOW%",
            current: "ends %ENDFROMNOW%",
            future: "starts %STARTFROMNOW% (%DURATION%)",
          },
          useEventTimeRelative: false,
          transform: (event)=>{
            if (event.title.search("Verjaardag") > -1) {
              event.icon = "fxemoji-partypopper"
            }
            if (event.title.search("RIP") > -1) {
              event.icon = "fxemoji-whitelatincross"
            }
            if (event.title.search("Trouwdag") > -1) {
              event.icon = "fxemoji-bouquetofflowers"
            }
            return event          },
        },
      ]
    },
  },
```

TestCal basic.ics :

```
BEGIN:VCALENDAR
PRODID:-//Google Inc//Google Calendar 70.9054//EN
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:PUBLISH
X-WR-CALNAME:TestCal
X-WR-TIMEZONE:Europe/Berlin
X-WR-CALDESC:Calendar for testing purposes
BEGIN:VEVENT
DTSTART;VALUE=DATE:20200528
DTEND;VALUE=DATE:20200529
DTSTAMP:20200527T110218Z
UID:54n3p18hquochdhe4f036o82rt@google.com
CREATED:20200527T105626Z
DESCRIPTION:
LAST-MODIFIED:20200527T105626Z
LOCATION:
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:FULLDAY_MAY_28
TRANSP:TRANSPARENT
END:VEVENT
BEGIN:VEVENT
DTSTART;VALUE=DATE:20200527
DTEND;VALUE=DATE:20200528
DTSTAMP:20200527T110218Z
UID:7rpm1sspsssn7i5bh3ci2fif4k@google.com
CREATED:20200527T105617Z
DESCRIPTION:
LAST-MODIFIED:20200527T105617Z
LOCATION:
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:FULLDAY_MAY_27
TRANSP:TRANSPARENT
END:VEVENT
BEGIN:VEVENT
DTSTART:20200528T180000Z
DTEND:20200528T190000Z
DTSTAMP:20200527T110218Z
UID:59789tvp4du292s0dhks8amn5e@google.com
CREATED:20200527T105543Z
DESCRIPTION:
LAST-MODIFIED:20200527T105543Z
LOCATION:
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:Second entry for TestCal
TRANSP:OPAQUE
END:VEVENT
BEGIN:VEVENT
DTSTART:20200527T180000Z
DTEND:20200527T190000Z
DTSTAMP:20200527T110218Z
UID:30318rf5j1ti1g3d0br5mio4e9@google.com
CREATED:20200527T105449Z
DESCRIPTION:
LAST-MODIFIED:20200527T105449Z
LOCATION:
SEQUENCE:0
STATUS:CONFIRMED
SUMMARY:First entry for TestCal
TRANSP:OPAQUE
END:VEVENT
END:VCALENDAR
```

**Comment by klaernie** [comment]
Thinking about this I can only imagine that it gets dropped by the view (maybe a timezone problem). I’ll have to get the debugger out to test this.

**Comment by klaernie** [comment]
Sorry, I must have been really tired while reading this issue the first two times. Now I noticed what the problem is.

You are absolutely correct - the upcoming view will not show events, that are currently in progress.
For an all-day event the start is right at midnight.
The upcoming view only shows events where the start time is in the future - but for the all-day event that date is in the past.

You might do the same as I do: add a `mode: "current"` view above your upcoming, and maybe hide the view when there are no events currently in progress:
```css
.CX2 .current .eventCount_0 {
	display: none;
}
```

**Comment by klaernie** [comment]
Also: thanks a lot for the many details in the report! Makes it much easier to understand the problem!

**Comment by evroom** [comment]
First of all thanks for taking the time to look at it.
Now I see that my actual issue is that I do not understand good what is "current" and "upcoming".
Do not see an explanation in the documents, or I did not look good enough.

I already used mode "upcoming", so I added a mode "current"'.
This indeed shows the current full day event(s) for today.
The "upcoming" events are the events for today (except for the full day event(s) for today) and future events (including full day event(s)).

<img width="797" alt="MMM-CalendarExt2 - screenshot 001" src="https://user-images.githubusercontent.com/40488822/83416365-d81e6200-a420-11ea-9b4a-b9777a3dd7bb.png">

So, better, but not perfect.

Where can I find an explanation of modes "current" and "upcoming" ?
